### DISCUSSION TITLE
'defineProps' is not defined

### DISCUSSION THREAD

**Discussion by Minori-ty** [eslint/eslint#15015] [Q&A] [answered]
***I use vue3, and defineProps don't need to import anymore.
But eslint report error***

```
//.eslintrc.js

module.exports = {
    root: true,
    env: {
        browser: true,
        node: true,
    },
    plugins: ['prettier'],
    extends: ['plugin:vue/vue3-essential', 'eslint:recommended', '@vue/prettier', 'prettier'],
    parserOptions: {
        parser: 'babel-eslint',
    },
    rules: {
        'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
        'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
        'prettier/prettier': [
            'warn',
            {
                singleQuote: true,
                semi: false,
            },
        ],
        indent: [
            2,
            4,
            {
                SwitchCase: 1,
            },
        ],
    },
}
```

**Comment by nzakas** [comment]
There’s not enough info here to be able to help you. However, if the rule throwing the error begins with “vue/“ then you’ll need to talk to the maintainers of eslint-plugin-vue for help.

**Comment by TNortnern** [comment, marked as answer]
any updates on this?

**Reply by huybn5776** [reply]
Add `defineProps` as global import.
[https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule](https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule)

**Reply by Minori-ty** [reply]
> Add `defineProps` as global import.
> https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule

Thank you， it work

**Reply by Miguel-Bento-Github** [reply]
Thank you for the help, this helped me find the solution.

The previous references don't work anymore and have been moved here.
https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-generate-no-undef-warnings

**Reply by divStar** [reply]
Excuse me.. so we still have to add the imports even though apparently they are not necessary?

I tried adding something along the following lines, but somehow it was successful once - and not anymore...
```js
  extends: [
    ...
    "plugin:vue/base",
    ...
  ],
  rules: {
    ...
    // Enable vue/script-setup-uses-vars rule
    'vue/script-setup-uses-vars': 'error',
    ...
  },
```
I don't know why it refuses to work anymore, but at some point it did work - without me adding any global or other imports anywhere.

Does anyone have an idea of how I accomplished it?

**Reply by KarmaBlackshaw** [reply]
Somehow the link provided by @huybn5776 returns the home page of eslint. Should we just add the `defineProps` as a global ?

**Reply by huybn5776** [reply]
> 
Its on [**Compiler macros such as defineProps and defineEmits are warned by no-undef rule**](https://eslint.vuejs.org/user-guide/#faq) section.
```js
// eslint.js
module.exports = {
  globals: {
    defineProps: "readonly",
    defineEmits: "readonly",
    defineExpose: "readonly",
    withDefaults: "readonly"
  }
}
```
